support/Python/Schedule.py

```diff
@@ -0,0 +1,754 @@
+        return range(int(match.group(1)), int(match.group(2)) + 1)
+    # Integers
+    match = re.match(r'(-?\d+)$', value)
+    if match:
+        logger.debug(f"'{value}' is an int")
+        return int(match.group(1))
+    # Floats
+    match = re.match(r'(-?\d+[.]\d*)$', value)
+    if match:
+        logger.debug(f"'{value}' is a float")
+        return float(match.group(1))
+    return value
+
+
+def _parse_params(ctx, param, all_values):
+    if all_values is None:
+        return {}
+    params = {}
+    for value in all_values:
+        key_and_value = value.split('=')
+        if len(key_and_value) != 2:
+            raise click.BadParameter(
+                f"The value of '{value}' could not be parsed as a key-value "
+                "pair. It should have a single '=' or none.")
+        params[key_and_value[0]] = _parse_param(key_and_value[1])
+    return params
+
+
+@click.command(help=schedule.__doc__.replace("**kwargs", "--params"))
+@click.argument('input_file_template',
+                type=click.Path(exists=True,
+                                file_okay=True,
+                                dir_okay=False,
+                                readable=True,
+                                path_type=Path))
+@click.option(
+    '--executable',
+    '-e',
+    show_default="executable listed in input file",
+    help=("The executable to run. Can be a path, or just the name "
+          "of the executable if it's in the 'PATH'. "
+          "If unspecified, the 'Executable' listed in the input file metadata "
+          "is used."))
+@click.option(
+    '--run-dir',
+    '-o',
+    # No `type=click.Path` because this can be a Jinja template
+    help=("The directory to which input file, submit script, etc. are "
+          "copied, relative to which the executable will run, and to "
+          "which output files are written. "
+          "Defaults to the current working directory if the input file is "
+          "already there and no scheduler is used. "
+          "Mutually exclusive with '--segments-dir' / '-O'."))
+@click.option(
+    '--segments-dir',
+    '-O',
+    # No `type=click.Path` because this can be a Jinja template
+    help=("The directory in which to create the next segment. "
+          "Requires '--from-checkpoint' or '--from-last-checkpoint' "
+          "unless starting the first segment."))
+@click.option('--copy-executable/--no-copy-executable',
+              default=None,
+              help=("Copy the executable to the run or segments directory. "
+                    "Default (when no flag is specified) is to copy only to "
+                    "the segments directory to support resubmission. "
+                    "With '--copy-executable' the executable is also copied "
+                    "to the run directory (only has an effect if "
+                    "'--run-dir' / '-o' is set). "
+                    "With '--no-copy-executable' the executable is never "
+                    "copied."))
```

Comment:
This is still a bit confusing. May I suggest

```
Copy the executable to the run or segments directory.
When no flag is specified:
 - If '--run-dir' / '-o' is set, don't copy.
 - If '-segments-dir' / '-O' is set, copy to segments directory to
   support resubmission.
If '--copy-executable' is specified:
 - If '--run-dir' / '-o' is set, copy to the run directory.
 - If '-segments-dir' / '-O' is set, copy to segments directory to
   support resubmission. (Note: the executable is not copied to
   individual segments)
If '--no-copy-executable' is specified: Never copy.
```

I know it's a bit verbose, but I think this is very clear on what the behavior is and is easy for users to follow.

Reply:
Sounds good, done.